images/management-api-for-apache-cassandra/tests/run-tests.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+set -o errexit -o nounset -o pipefail
+
+# Start a Cassandra container
+CASSANDRA_CONTAINER=$(docker run -d -e CASSANDRA_START_RPC=true $IMAGE_NAME)
+
+# Wait for Cassandra to start
+MAX_RETRIES=10
+RETRY_INTERVAL=10
+
+for ((i=0; i<MAX_RETRIES; i++)); do
+  if docker exec "$CASSANDRA_CONTAINER" nodetool status; then
+    echo "Cassandra is running"
+    break
+  else
+    echo "Waiting for Cassandra to start..."
+    sleep $RETRY_INTERVAL
+  fi
+
+  if [[ $i -eq $((MAX_RETRIES-1)) ]]; then
+    echo "Error: Cassandra did not start within the expected time"
+    docker logs "$CASSANDRA_CONTAINER"
+    docker stop "$CASSANDRA_CONTAINER"
+    docker rm "$CASSANDRA_CONTAINER"
+    exit 1
+  fi
+done
+
+# Create a keyspace and table, insert data, and query it
+for ((i=0; i<MAX_RETRIES; i++)); do
+  if docker exec -i "$CASSANDRA_CONTAINER" cqlsh <<EOF
+CREATE KEYSPACE testkeyspace WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};
+USE testkeyspace;
+CREATE TABLE users (user_id UUID PRIMARY KEY, name text);
+INSERT INTO users (user_id, name) VALUES (uuid(), 'Chainguard');
+SELECT * FROM users;
+EOF
+  then
+    echo "Keyspace and table created, data inserted and queried successfully"
+    break
+  else
+    echo "Waiting for Cassandra to be ready for CQL commands..."
+    sleep $RETRY_INTERVAL
+  fi
+
+  if [[ $i -eq $((MAX_RETRIES-1)) ]]; then
+    echo "Error: Cassandra did not accept CQL commands within the expected time"
+    docker logs "$CASSANDRA_CONTAINER"
+    docker stop "$CASSANDRA_CONTAINER"
+    docker rm "$CASSANDRA_CONTAINER"
+    exit 1
+  fi
+done
+
+# Check if the data was inserted and queried correctly
+for ((i=0; i<MAX_RETRIES; i++)); do
+  RESULT=$(docker exec "$CASSANDRA_CONTAINER" cqlsh -e "USE testkeyspace; SELECT * FROM users;")
+  if [[ "$RESULT" == *"Chainguard"* ]]; then
+    echo "Data insertion/query successful"
+    break
+  else
+    echo "Waiting for data to be available in the table..."
+    sleep $RETRY_INTERVAL
+  fi
+
+  if [[ $i -eq $((MAX_RETRIES-1)) ]]; then
+    echo "Error: Data was not available in the table within the expected time"
+    docker logs "$CASSANDRA_CONTAINER"
+    docker stop "$CASSANDRA_CONTAINER"
+    docker rm "$CASSANDRA_CONTAINER"
+    exit 1
+  fi
+done
+
+# Clean up
+docker stop "$CASSANDRA_CONTAINER"
+docker rm "$CASSANDRA_CONTAINER"
```

Comment:
```suggestion
docker rm -f "$CASSANDRA_CONTAINER"
```

fishing in the dark, I wonder if this is racing?

Reply:
valid shout-out, actually we don't need these at all now the test is using the test harness right? so i'll remove both and not do any cleanup in the test and see how far that gets

Reply:
didn't make a difference unfortunately but continuing to try a few more things